### ISSUE TITLE
Internal servlets are available on external context

### ISSUE THREAD

**Issue by davidmc24** [wotifgroup/grails-lightweight-deploy#30]
Steps to reproduce:
- Follow the instructions in the documentation under "Customised Bootstrapping" to create a custom `ApplicationLauncher`.
- Override `configureInternalServlets` to register an internal servlet (example below).
- Package and run the application.
- Expected behavior: internal servlet is not accessible on the external port (in my case, `8080`)
- Actual behavior: internal servlet is accessible on the external port (in my case, `8081`)

It looks like the problem is with the `handler.setConnectorNames` call in `createInternalContext` and `createExternalContext`, which attach the handler to all connectors that exist at that point in the startup process.

```
    @Override
    protected void configureInternalServlets(ServletContextHandler handler) {
        super.configureInternalServlets(handler);
        handler.addServlet(new ServletHolder(new InfoServlet()), "/info/*");
    }
```

**Comment by davidmc24** [comment]
This may not normally be visible.  However, it is if the external context has a non-root context path set.

**Comment by deverton** [comment]
Closed by #40
